src/libraries/Microsoft.Extensions.Configuration.Binder/tests/Common/ConfigurationBinderTests.Collections.cs

```diff
@@ -815,11 +817,14 @@ public void NonStringKeyDictionaryBinding()
 
             var options = new OptionsWithDictionary();
             config.Bind(options);
-
+#if BUILDING_SOURCE_GENERATOR_TESTS // Source generator will not touch the property if it is not supported.
```

Comment:
>// Source generator will not touch the property if it is not supported. [](http://example.com/codeflow?start=36&length=71)

Is this temporary or you are going to keep this behavior?

Reply:
I'd consider this a divergence. It seems wasteful to allocate an object if we can't populate it.